bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/WebDataDownloadUtil.java

```diff
@@ -0,0 +1,148 @@
+     *
+     * If content type is not found in the headers, the data is scanned to determine the content type.
+     *
+     * @param url the URL of the image to be downloaded
+     * @return a RawType object containing the image, null if the content type could not be found or the content type is
+     *         not an image
+     */
+    public static RawType downloadImage(URL url) {
+        return WebDataDownloadUtil.downloadImage(url, true, -1);
+    }
+
+    /**
+     * Download the image data from an URL.
+     *
+     * @param url the URL of the image to be downloaded
+     * @param scanTypeInContent true to allow the scan of data to determine the content type if not found in the headers
+     * @param maxContentLength the maximum data size in bytes to trigger the download; any negative value to ignore the
+     *            data size
+     * @return a RawType object containing the image, null if the content type could not be found or the content type is
+     *         not an image or the data size is too big
+     */
+    public static RawType downloadImage(URL url, boolean scanTypeInContent, long maxContentLength) {
+        return WebDataDownloadUtil.downloadData(url, "image/.*", scanTypeInContent, maxContentLength);
+    }
+
+    /**
+     * Download the data from an URL.
+     *
+     * @param url the URL of the data to be downloaded
+     * @param contentTypeRegex the REGEX the content type must match; null to ignore the content type
+     * @param scanTypeInContent true to allow the scan of data to determine the content type if not found in the headers
+     * @param maxContentLength the maximum data size in bytes to trigger the download; any negative value to ignore the
+     *            data size
+     * @return a RawType object containing the downloaded data, null if the content type does not match the expected
+     *         type or the data size is too big
+     */
+    public static RawType downloadData(URL url, String contentTypeRegex, boolean scanTypeInContent,
+            long maxContentLength) {
+
+        String contentType = RawType.DEFAULT_MIME_TYPE;
+        if (contentTypeRegex != null) {
+            contentType = WebDataDownloadUtil.getContentType(url, scanTypeInContent, maxContentLength);
+            if (contentType == null) {
+                LoggerFactory.getLogger(WebDataDownloadUtil.class)
+                        .debug("Unknown content type from URL {} ; download aborted", url.toExternalForm());
+                return null;
+            } else if (!contentType.matches(contentTypeRegex)) {
+                LoggerFactory.getLogger(WebDataDownloadUtil.class).debug(
+                        "Unexpected content type \"{}\" from URL {} ; download aborted", contentType,
+                        url.toExternalForm());
+                return null;
+            }
+        }
+
+        RawType rawData;
+        InputStream input = null;
+        try {
+            input = url.openStream();
```

Comment:
Hard to get it right, just think about proxies, handling all possible error situations, not leaking any resources... Therefore please use the available http client (jetty).